pkg/controller/clone-controller.go

```diff
@@ -91,11 +91,20 @@ func (cc *CloneController) findClonePodsFromCache(pvc *v1.PersistentVolumeClaim)
 func (cc *CloneController) processPvcItem(pvc *v1.PersistentVolumeClaim) error {
 	anno := map[string]string{}
 
+	deferFunc := func(cc *CloneController, anno map[string]string, sourcePod *v1.Pod, targetPod *v1.Pod) {
+		if anno[AnnCloneOf] == "true" {
```

Comment:
This check in the deferred deletePod call is what fixes the bug.  We work the opposite direction and we don't delete the pod until the annotation is set, so this way subsequent updates don't delete the pod before the annotation is set.

This might cause problems in other places, but it does fix up the bug.

Reply:
I guess I don't follow.  I understand that the pods would now be deleted after the pvc is updated.  And that may helps if say the controller crashes after the pods are deleted (and the pvc never gets updated and another controller takes over somehow).  But the controller isn't crashing as far as I know.  And the conditions that set the "cloneof" annotation are the same.  I don't see how deleting the pods a couple miliseconds after the pvc helps us in the non-failure case.

Reply:
@mhenriks Yeah, my understanding is that there's a race here and in a normal flow the pod is being deleted before the annotation is updated on the PVC, and that's what's being addressed here.